rclcpp/include/rclcpp/executors.hpp

```diff
@@ -26,6 +29,20 @@ namespace executors
 using rclcpp::executors::multi_threaded_executor::MultiThreadedExecutor;
 using rclcpp::executors::single_threaded_executor::SingleThreadedExecutor;
 
+template<typename FutureT>
+std::shared_future<FutureT> &
+spin_node_until_future_complete(
+  rclcpp::executor::Executor & executor, rclcpp::node::Node::SharedPtr & node_ptr,
+  std::shared_future<FutureT> & future)
+{
+  std::future_status status;
+  do {
+    executor.spin_node_some(node_ptr);
```

Comment:
It may be worth putting a TODO here mentioning that this won't work recursively right now, i.e. you can't call `spin_node_until_future_complete` inside a callback executed by an executor.